recipes/sail/all/patches/0.9.0-fix-libjxl-find.patch

```diff
@@ -0,0 +1,28 @@
+diff --git a/src/sail-codecs/jpegxl/CMakeLists.txt b/src/sail-codecs/jpegxl/CMakeLists.txt
+index 54441fa6..7c290456 100644
+--- a/src/sail-codecs/jpegxl/CMakeLists.txt
++++ b/src/sail-codecs/jpegxl/CMakeLists.txt
+@@ -1,3 +1,4 @@
++if(0)
+ find_library(JPEGXL_LIBRARY jxl                 NAMES jxl jxl-static                 ${SAIL_CODEC_JPEGXL_REQUIRED_OPTION})
+ find_library(JPEGXL_THREADS_LIBRARY jxl_threads NAMES jxl_threads jxl_threads-static ${SAIL_CODEC_JPEGXL_REQUIRED_OPTION})
+ find_path(JPEGXL_INCLUDE_DIRS jxl/decode.h                                           ${SAIL_CODEC_JPEGXL_REQUIRED_OPTION})
+@@ -50,3 +51,18 @@ sail_codec(NAME jpegxl
+             DEPENDENCY_INCLUDE_DIRS ${JPEGXL_INCLUDE_DIRS}
+             DEPENDENCY_LIBS ${BROTLI_COMMON_LIBRARY} ${BROTLI_DEC_LIBRARY} ${HWY_LIBRARY}
+                             ${JPEGXL_LIBRARY} ${JPEGXL_THREADS_LIBRARY})
++endif()
++# Find libjxl package
++find_package(libjxl REQUIRED CONFIG)
++
++# Common codec configuration
++sail_codec(NAME jpegxl
++           SOURCES helpers.h helpers.c jpegxl.c memory.h memory.c
++           ICON jpegxl.png)
++
++# For static builds, define JXL_STATIC_DEFINE
++if(NOT BUILD_SHARED_LIBS)
++    target_compile_definitions(sail-codec-jpegxl PRIVATE JXL_STATIC_DEFINE)
```

Comment:
`JXL_STATIC_DEFINE` should be defined (transitively) by the `libjxl::libjxl` target when _that_ target is built statically, so the line below (assuming `SAIL_CODEC_TARGET` is `sail-codec-jpegxl`) should already be handling this correctly? 

https://github.com/conan-io/conan-center-index/blob/879e140d9438ab491aceea766d0cf0ae3595dae8/recipes/libjxl/all/conanfile.py#L190